internal/runtimehandlerhooks/high_performance_hooks_linux.go

```diff
@@ -77,35 +79,78 @@ func (h *HighPerformanceHooks) PreCreate(ctx context.Context, specgen *generate.
 		return nil
 	}
 
+	cpusString := specgen.Config.Linux.Resources.CPU.Cpus
+
+	exclusiveCPUSet, err := cpuset.Parse(cpusString)
+	if err != nil {
+		return fmt.Errorf("failed to parse container %q cpus: %w", c.Name(), err)
+	}
+
+	sharedCPUSet, err := cpuset.Parse(h.sharedCPUs)
+	if err != nil {
+		return fmt.Errorf("failed to parse shared cpus: %w", err)
+	}
+
+	h.setExecCPUAffinity(ctx, specgen, &exclusiveCPUSet, &sharedCPUSet)
+
 	if requestedSharedCPUs(s.Annotations(), c.CRIContainer().GetMetadata().GetName()) {
 		if isContainerCPUsSpecEmpty(specgen.Config) {
 			return fmt.Errorf("no cpus found for container %q", c.Name())
 		}
 
-		cpusString := specgen.Config.Linux.Resources.CPU.Cpus
-
-		exclusiveCPUs, err := cpuset.Parse(cpusString)
-		if err != nil {
-			return fmt.Errorf("failed to parse container %q cpus: %w", c.Name(), err)
-		}
-
-		if h.sharedCPUs == "" {
+		if sharedCPUSet.IsEmpty() {
 			return fmt.Errorf("shared CPUs were requested for container %q but none are defined", c.Name())
 		}
 
-		sharedCPUSet, err := cpuset.Parse(h.sharedCPUs)
-		if err != nil {
-			return fmt.Errorf("failed to parse shared cpus: %w", err)
-		}
 		// We must inject the environment variables in the PreCreate stage,
 		// because in the PreStart stage the process is already constructed.
 		// by the low-level runtime and the environment variables are already finalized.
-		injectCpusetEnv(specgen, &exclusiveCPUs, &sharedCPUSet)
+		injectCpusetEnv(specgen, &exclusiveCPUSet, &sharedCPUSet)
 	}
 
 	return nil
 }
 
+// setExecCPUAffinity sets ExecCPUAffinity in the container spec.
+func (h *HighPerformanceHooks) setExecCPUAffinity(ctx context.Context, specgen *generate.Generator, exclusiveCPUSet, sharedCPUSet *cpuset.CPUSet) cpuset.CPUSet {
```

Comment:
The logic looks good to me. But why didn't you add a unit test for setExecCPUAffinity and also extend the unit test for PreCreate to cover the new functionality you added?